### PR TITLE
e2e ErrorCollector test scenarios

### PR DESCRIPTION
**Pull request by sawantpritam**
@rmstar @itsankit-google Please review.

### CODE REVIEW

**Comment by itsmekumari** [comment]
@itsankit-google The tests are passed, after re-trigger using the latest e2e-framework of locators changes.

File: core-plugins/src/e2e-test/resources/pluginParameters.properties [hunk 1]
```diff
@@ -185,3 +185,12 @@ groupByTest2OutputFile=testdata/expected_outputs/CSV_GROUPBY_TEST2_Output.csv
 groupByTest3OutputFile=testdata/expected_outputs/CSV_GROUPBY_TEST3_Output.csv
 groupByMacroOutputFile=testdata/expected_outputs/CSV_GROUPBY_TEST4_Output.csv
 ## GROUPBY-PLUGIN-PROPERTIES-END
+
+## ERROR-COLLECTOR-PROPERTIES-START
+errorMessageColumnName=errorMessage
+errorCodeColumnName=errorCode
+errorEmitterNodeName=errorEmitterNode
+errorCollectorWranglerCondition=send-to-error workhours < 9 \nsend-to-error Salary > 40000
+errorCollectorDefaultOutput=testdata/expected_outputs/CSV_ERROR_COLLECTOR_DEFAULT_Output.csv
```
Comment (itsankit-google) [inline core-plugins/src/e2e-test/resources/pluginParameters.properties]: nit - `errorCollectorDefaultConfigOutput` & `CSV_ERROR_COLLECTOR_DEFAULT_CONFIG_Output.csv`

File: core-plugins/src/e2e-test/resources/pluginParameters.properties [hunk 2]
```diff
@@ -185,3 +185,12 @@ groupByTest2OutputFile=testdata/expected_outputs/CSV_GROUPBY_TEST2_Output.csv
...
+errorCollectorConfigOutput=testdata/expected_outputs/CSV_ERROR_COLLECTOR_CONFIG_Output.csv
```
Comment (itsankit-google) [inline core-plugins/src/e2e-test/resources/pluginParameters.properties]: nit - `errorCollectorCustomConfigOutput` & `CSV_ERROR_COLLECTOR_CUSTOM_CONFIG_Output.csv`

File: core-plugins/src/e2e-test/features/errorcollector/ErrorCollector.feature [hunk 3]
```diff
@@ -71,3 +71,110 @@ Feature: Error Collector - Verification of successful error emitted to ErrorColl
     Then Verify the pipeline status is "Succeeded"
     Then Verify the CSV Output File matches the Expected Output File: "errorCollectorOutput" With Expected Partitions: "expectedErrorCollectorOutputPartitions"
     Then Close the pipeline logs
+
+  @FILE_SOURCE_TEST @FILE_SINK_TEST @ErrorCollector
+  Scenario: Verify error collector plugin functionality with default error collector config using File source to Wrangler to ErrorCollector to File sink pipeline
```
Comment (itsankit-google) [inline core-plugins/src/e2e-test/features/errorcollector/ErrorCollector.feature]: nit - `Verify error collector plugin functionality with default config using File Plugins and Wrangler`

File: core-plugins/src/e2e-test/features/errorcollector/ErrorCollector.feature [hunk 4]
```diff
@@ -71,3 +71,110 @@ Feature: Error Collector - Verification of successful error emitted to ErrorColl
...
+    Then Close the preview
+    Then Save and Deploy Pipeline
+    Then Run the Pipeline in Runtime
+    Then Wait till pipeline is in running state
+    Then Open and capture logs
+    Then Verify the pipeline status is "Succeeded"
+    Then Validate output records in output folder path "filePluginOutputFolder" is equal to expected output file "errorCollectorDefaultOutput"
+
+  @FILE_SOURCE_TEST @FILE_SINK_TEST @ErrorCollector
+  Scenario: Verify error collector plugin functionality by overriding default error collector config using File source to Wrangler to ErrorCollector to File sink pipeline
```
Comment (itsankit-google) [inline core-plugins/src/e2e-test/features/errorcollector/ErrorCollector.feature]: nit - `Verify error collector plugin functionality with custom config using File Plugins and Wrangler`

File: core-plugins/src/e2e-test/resources/pluginParameters.properties (hunk 1 above)
Comment (sawantpritam) [reply core-plugins/src/e2e-test/resources/pluginParameters.properties]: Changed.

File: core-plugins/src/e2e-test/resources/pluginParameters.properties (hunk 2 above)
Comment (sawantpritam) [reply core-plugins/src/e2e-test/resources/pluginParameters.properties]: Renamed.

File: core-plugins/src/e2e-test/features/errorcollector/ErrorCollector.feature (hunk 3 above)
Comment (sawantpritam) [reply core-plugins/src/e2e-test/features/errorcollector/ErrorCollector.feature]: Changed description

File: core-plugins/src/e2e-test/features/errorcollector/ErrorCollector.feature (hunk 4 above)
Comment (sawantpritam) [reply core-plugins/src/e2e-test/features/errorcollector/ErrorCollector.feature]: Changed description

**Review by itsankit-google — APPROVED** [review]

**Pull request merged**